modules/mt/keeper/keeper.go

```diff
@@ -48,40 +48,62 @@ func (k Keeper) IssueDenom(ctx sdk.Context,
 }
 
 // MintMT mints an MT and manages the MT's existence within Collections and Owners
-func (k Keeper) MintMT(
-	ctx sdk.Context, denomID, tokenID string, amout uint64, data []byte, owner sdk.AccAddress,
+func (k Keeper) MintMT(ctx sdk.Context,
+	denomID, tokenID string,
+	amout uint64,
+	data []byte,
+	owner sdk.AccAddress,
 ) error {
 
-	// TODO create if not exists, mint if exists
 	if k.HasMT(ctx, denomID, tokenID) {
-		return sdkerrors.Wrapf(types.ErrMTAlreadyExists, "MT %s already exists in collection %s", tokenID, denomID)
+		mt, err := k.GetMT(ctx, denomID, tokenID)
+		if err != nil {
+			return err
+		}
+
+		k.setMT(
+			ctx, denomID,
+			types.MT{
+				Id:     mt.GetID(),
+				Supply: amout + mt.GetSupply(),
+				Data:   owner,
+				Owner:  mt.GetOwner().String(),
+			},
+		)
+	} else {
+		k.setMT(
+			ctx, denomID,
+			types.NewMT(
+				tokenID,
+				amout,
+				owner,
+				data,
+			),
+		)
+
+		k.setOwner(ctx, denomID, tokenID, amout, owner)
+		// todo 确定是否还需要 collection
+		k.increaseSupply(ctx, denomID)
 	}
 
-	k.setMT(
-		ctx, denomID,
-		types.NewMT(
-			tokenID,
-			amout,
-			owner,
-			data,
-		),
-	)
-	k.setOwner(ctx, denomID, tokenID, owner)
-	k.increaseSupply(ctx, denomID)
-
 	return nil
 }
 
 // EditMT updates an existing MT
-func (k Keeper) EditMT(
-	ctx sdk.Context, denomID, tokenID string, tokenData []byte, owner sdk.AccAddress,
+func (k Keeper) EditMT(ctx sdk.Context,
+	denomID, tokenID string,
+	tokenData []byte,
+	owner sdk.AccAddress,
 ) error {
-	_, found := k.GetDenom(ctx, denomID)
+	denom, found := k.GetDenom(ctx, denomID)
 	if !found {
 		return sdkerrors.Wrapf(types.ErrInvalidDenom, "Denom not found: %s", denomID)
 	}
 
-	// TODO only the owner of Denom can edit
+	if denom.Owner != owner.String() {
+		return sdkerrors.Wrapf(types.ErrUnauthorized, "Denom is owned by %s", denom.Owner)
+	}
+
 	mt, err := k.Authorize(ctx, denomID, tokenID, owner)
```

Comment:
`Authorize` checks if the sender is the owner of the given `MT`, but not of the given `denom`, so I guess here is a bug when the denom owner is not the MT owner.

Maybe we do not need the `owner` property in the `MT` structure, because the `MT` is owned by the `denom`